tools/twix/src/panels/image_color_select.rs

```diff
@@ -0,0 +1,282 @@
+                grid.show(ui, |ui| {
+                    ui.label(RichText::new("max:").strong());
+                    ui.colored_label(Color32::RED, format!("r: {:.3}", max.red));
+                    ui.colored_label(Color32::GREEN, format!("g: {:.3}", max.green));
+                    ui.colored_label(
+                        Color32::from_rgb(50, 150, 255),
+                        format!("b: {:.3}", max.blue),
+                    );
+                    ui.end_row();
+
+                    ui.label(RichText::new("min:").strong());
+                    ui.colored_label(Color32::RED, format!(" r: {:.3}", min.red));
+                    ui.colored_label(Color32::GREEN, format!("g: {:.3}", min.green));
+                    ui.colored_label(
+                        Color32::from_rgb(50, 150, 255),
+                        format!("b: {:.3}", min.blue),
+                    );
+                    ui.end_row();
+
+                    ui.label(RichText::new("average:").strong());
+                    ui.colored_label(Color32::RED, format!(" r: {:.3}", average.red));
+                    ui.colored_label(Color32::GREEN, format!("g: {:.3}", average.green));
+                    ui.colored_label(
+                        Color32::from_rgb(50, 150, 255),
+                        format!("b: {:.3}", average.blue),
+                    );
+                    ui.end_row();
+                });
+
+                painter.circle(
+                    pixel_pos,
+                    self.brush_size,
+                    Color32::TRANSPARENT,
+                    Stroke::new(1.0, Color32::BLACK),
+                );
+            }
+        }
+        response
+    }
+}
+
+impl ImageColorSelectPanel {
+    fn get_image(&self) -> Result<ColorImage> {
+        let image_data: YCbCr422Image = self
+            .image_buffer
+            .parse_latest()
+            .map_err(|error| eyre!("{error}"))?;
+        let buffer = image_data
+            .buffer()
+            .iter()
+            .flat_map(|&ycbcr422| {
+                let ycbcr444: [YCbCr444; 2] = ycbcr422.into();
+                ycbcr444
+            })
+            .flat_map(|ycbcr444| {
+                let rgb = Rgb::from(ycbcr444);
+                [rgb.r, rgb.g, rgb.b, 255]
+            })
+            .collect::<Vec<_>>();
+        let image = ColorImage::from_rgba_unmultiplied([640, 480], &buffer);
+        Ok(image)
+    }
+}
+
+fn get_pixel_chromaticity(image: &ColorImage, pixel_pos: Point2<Pixel>) -> PixelColor {
+    let color32 = image.pixels[(pixel_pos.y() as usize) * image.width() + (pixel_pos.x() as usize)];
+    let sum = color32.r() as f32 + color32.g() as f32 + color32.b() as f32;
+    let mut pixel = PixelColor {
+        red: 0.0,
+        green: 0.0,
+        blue: 0.0,
+    };
+    if sum != 0.0 {
+        pixel.red = (color32.r() as f32) / sum;
+        pixel.green = (color32.g() as f32) / sum;
+        pixel.blue = (color32.b() as f32) / sum;
+    }
+    pixel
+}
```

Comment:
The `Rgb` struct already has such an implementation.

Reply:
same as in the previous, this is RGB and no rgb